v2/pkg/genruntime/secrets/collector.go

```diff
@@ -115,11 +116,12 @@ func (c *Collector) Values() ([]*v1.Secret, error) {
 	result := maps.Values(c.secrets)
 
 	// Force a deterministic ordering
-	sort.Slice(result, func(i, j int) bool {
-		left := result[i]
-		right := result[j]
+	slices.SortFunc(result, func(left, right *v1.Secret) int {
+		if c := cmp.Compare(left.Namespace, right.Namespace); c != 0 {
+			return c
+		}
 
-		return left.Namespace < right.Namespace || (left.Namespace == right.Namespace && left.Name < right.Name)
+		return cmp.Compare(left.Name, right.Name)
```

Comment:
Curious what the reason for the sortFunc change here is?

I'm not understanding what the difference is between what we had and what's here now, in terms of actual behavior.

Reply:
I think we've both changed this. You added the ordering by namespace for packages with identical names; I updated it to use the new API. I'll update it.